app/src/main/kotlin/nl/eduid/screens/personalinfo/PersonalInfoScreen.kt

```diff
@@ -202,102 +200,172 @@ fun PersonalInfoScreen(
         if (!isLoading) {
             if (!uiState.personalInfo.isVerified) {
                 NotVerifiedIdentity(
-                    selfAssertedName = uiState.personalInfo.seflAssertedName,
                     isLoading = isLoading,
                     addLinkToAccount = addLinkToAccount,
-                    onNameClicked = onNameClicked
                 )
             } else {
-                VerifiedIdentity(
-                    personalInfo = uiState.personalInfo,
-                    onNameClicked = onNameClicked,
-                    openVerifiedInformation = openVerifiedInformation
-                )
+                VerifiedIdentity()
             }
-
-            Text(
-                style = MaterialTheme.typography.titleLarge.copy(color = MaterialTheme.colorScheme.onSecondary),
-                text = stringResource(R.string.Profile_ContactDetails_COPY),
-                modifier = Modifier.padding(top = 16.dp)
-            )
-            InfoField(title = uiState.personalInfo.email,
-                subtitle = stringResource(R.string.Profile_Email_COPY),
+            val personalInfo = uiState.personalInfo
+            // We always allow editing the first name
+            InfoField(
+                title = personalInfo.selfAssertedName.chosenName.orEmpty(),
+                subtitle = stringResource(R.string.Profile_FirstName_COPY),
                 modifier = Modifier.clickable {
-                    onEmailClicked()
-                })
-
-            if (uiState.personalInfo.institutionAccounts.isNotEmpty()) {
-                RoleAndInstitutions(openVerifiedInformation, uiState.personalInfo.institutionAccounts)
-            }
-
-            LinkAccountCard(
-                title = R.string.Profile_AddRoleAndInstitution_COPY,
-                subtitle = R.string.Profile_AddViaSurfconext_COPY,
-                addLinkToAccount = addLinkToAccount
+                    onNameClicked()
+                }
             )
-            val configuration = LocalConfiguration.current
-            Surface(
-                modifier = Modifier
-                    .requiredWidth(configuration.screenWidthDp.dp)
-                    .background(MaterialTheme.colorScheme.tertiaryContainer)
-                    .padding(vertical = 12.dp, horizontal = 24.dp),
-            ) {
-                OutlinedButton(
-                    onClick = onManageAccountClicked,
-                    shape = RoundedCornerShape(CornerSize(6.dp)),
-                    modifier = Modifier
-                        .sizeIn(minHeight = 48.dp)
-                        .fillMaxWidth()
-                ) {
-                    Image(
-                        painter = painterResource(R.drawable.cog_icon),
-                        alignment = CenterStart,
-                        contentDescription = "",
-                        modifier = Modifier.padding(horizontal = 24.dp)
+
+            var hasVerifiedGivenName = false
```

Comment:
This variable checks if we have already added a first name / last name.
We know what the name selected by the user is, but we don't know yet which linked account it comes from.
So until we have not found the given link, we go through each of the accounts, and search for it.
If this variable is so to true, we stop with searching, because we just added the UI for it.
I can move this search to the ViewModel if needed.